main.c

```diff
@@ -0,0 +1,572 @@
+            }
+            if (found) {
+                break; // Exit the outer loop as we have found the student
+            }
+        }
+        if (found) {
+            break; // Exit the outermost loop as we have found the student
+        }
+    }
+
+    if (!found) {
+        printf("Student with phone number %s not found.\n", phoneNumber);
+    }
+}
+
+Student searchStudent(char firstName[MAXNUMF],char lastName[MAXNUML]) {
+    
+    
+    Student* listofstudents[CLASS*LEVELS];
+    int found = 0;
+
+   
+    // Traverse the entire school array to find the student with the given first name and last name
+    for (int level = 0; level < LEVELS; level++) {
+        for (int class = 0; class < CLASS; class++) {
+            Student* p = school[level][class];
+            while (p != NULL) {
+                if (strcmp(p->firstName, firstName) == 0 && strcmp(p->lastName, lastName) == 0) {
+                    listofstudents[found]=p;
+                    found++;
+                    printf("Name: %s %s\n", p->firstName, p->lastName);
+                    printf("Phone Number: %s\n", p->phoneNumber);
+                    printf("Level: %d\n", level + 1);
+                    printf("Class: %d\n", class + 1);
+                    printf("Grades: ");
+                    for (int i = 0; i < 10; i++) {
+                        printf("%d ", p->grades[i]);
+                    }
+                    printf("\n");
+
+                    
+                }
+                p = p->next;
+            }
+            
+        }
+        
+    }
+    int index=0;
+    if (!found) {
+        printf("Student with first name %s and last name %s not found.\n", firstName, lastName);
+        return;
+    }else if(found>1){
+        printf("There are some students with this name. choose one of them from the list above with index between 0 to %d\n",found);
+        do{
+        printf("Enter the number of index: ");
+        scanf("%d", &index);
+        }while(index<0||index>found);
+    }
+    return *listofstudents[index];
+}
+
+
+void printTopNStudentsPerCourse() {
+     int subjectIndex;
+     int numberOfStudent=10;
+    do{
+        printf("Enter the number of subject(1-10): ");
+        scanf("%d", &subjectIndex);
+    }while(subjectIndex<1||subjectIndex>10);
+    
+
+    // Create an array to store the top 10 outstanding students for each class
+    Student* topStudents[LEVELS][numberOfStudent]; // Assuming each class can have up to 10 outstanding students
+    int numTopStudents[LEVELS] = {0}; // Number of outstanding students in each class
+
+    // Traverse the entire school array and find the top 10 outstanding students for each class
+    for (int level = 0; level < LEVELS; level++) {
+        for (int class = 0; class < CLASS; class++) {
+            Student* p = school[level][class];
+            while (p != NULL) {
+               
+                // Sort the students by the given subject's grades in descending order
```

Comment:
general question:
Why sort is needed here?
What the complexity will be without sort?